test/index.ts

```diff
@@ -1258,6 +1258,20 @@ const commentSchemaObject = Joi.object<Comment, true>({
     user: userSchemaObject
 });
 
+interface Comment2 {
+    text: string;
+    user: {
```

Comment:
For accuracy, this doesn't change the test but it matches what joi will actually convert it to:
```suggestion
    user?: {
```